packages/salesforcedx-vscode-core/src/commands/baseDeployRetrieve.ts

```diff
@@ -232,10 +239,19 @@ export abstract class RetrieveExecutor<T> extends DeployRetrieveExecutor<T> {
     this.setupCancellation(operation, token);
 
     const result: RetrieveResult = await operation.pollStatus();
-    await SourceTrackingService.updateSourceTrackingAfterRetrieve(
-      sourceTracking,
-      result
-    );
+
+    if (
+      result &&
+      result.response &&
+      (result.response.status === 'Succeeded' ||
+        result.response.status === 'SucceededPartial') &&
```

Comment:
nit: you could use [optional chaining](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html) here 
```
if  ((result?.response?.status === 'Succeeded' || result?.response?.status === 'SucceededPartial') && this.sourceTracking) {
```
or slightly cleaner 
```
const status = result?.response?.status;
if  ((status === 'Succeeded' || status === 'SucceededPartial') && this.sourceTracking) {
```